### DISCUSSION TITLE
Fehler obwohl Funktion ok?

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2112] [Q&A] [answered]
Datenpunkt-Eigenschaften Umrechnungsversuch:
`{"value":"val =>  {let ms = new Date().getTime()-(new Date(val.substr(0,4)+'-'+val.substr(5,2)+'-'+val.substr(8,2)+' '+val.substr(11,8)).getTime()); let d, h, m, s;  s = Math.floor(ms / 1000);  m = Math.floor(s / 60);  s = s % 60;  h = Math.floor(m / 60);  m = m % 60;  d = Math.floor(h / 24);  h = h % 24; let withZ =function (n) { return n < 10 ? '0' + n : n; }; return d+ ' Tag(e) '+ withZ(h) + ':' + withZ(m) + ':' + withZ(s); } "}`

Ausgangswert
![image](https://user-images.githubusercontent.com/84818934/208546928-2caa0e5f-f930-4757-855d-94a30d8a8f26.png)

`2022-12-17T20:08:03`

Type TSval
![image](https://user-images.githubusercontent.com/84818934/208547220-319d71f4-6d7d-4acf-8280-adde02f3cb55.png)

Ergebnis wird korrekt angezeigt:

![image](https://user-images.githubusercontent.com/84818934/208546870-b6f5ea96-2099-4ad5-8020-ce05c83230ce.png)

Auch mit `substring` und entsprechend angepassten Werten kommt der Fehler.

![image](https://user-images.githubusercontent.com/84818934/208546716-0aa89d20-84e5-4bcb-bdc2-34e536981668.png)

**Comment by Zefau** [comment, marked as answer]
Beim Ladevorgang kann es sein, dass `val` keinen Wert hat, also `undefined` ist. Das musst du abfangen:

```
{"value":"val => {if (!val) {return '';} let ms = new Date().getTime()-(new Date(val.substr(0,4)+'-'+val.substr(5,2)+'-'+val.substr(8,2)+' '+val.substr(11,8)).getTime()); let d, h, m, s;  s = Math.floor(ms / 1000);  m = Math.floor(s / 60);  s = s % 60;  h = Math.floor(m / 60);  m = m % 60;  d = Math.floor(h / 24);  h = h % 24; let withZ =function (n) { return n < 10 ? '0' + n : n; }; return d+ ' Tag(e) '+ withZ(h) + ':' + withZ(m) + ':' + withZ(s); } "}
```
